ansible/roles/atmo-ssh-setup/tasks/main.yml

```diff
@@ -102,6 +102,33 @@
 ###
 # OTHER SSH CONFIG CHANGES
 ##
+
+- name: manual OS detection
+  stat: path=/etc/redhat-release
+  register: os_ver
+  tags:
+    - debug
+    - ssh-restart
+
+- name: /etc/ssh/sshd_config changes
+  lineinfile: dest=/etc/ssh/sshd_config backup=yes regexp="{{ item.regexp }}" line="{{ item.line }}"
+  with_items:
+    - { regexp: '^PermitRootLogin', line: 'PermitRootLogin without-password'}
+    - { regexp: '^UseDNS', line: 'UseDNS no'}
+
+  #if port line is commented or not commented out it is removed
+- name: ensure that the ports are correct in sshd_config
+  lineinfile: dest=/etc/ssh/sshd_config backup=yes regexp="^Port" state=absent
+- lineinfile: dest=/etc/ssh/sshd_config backup=yes regexp="^#Port" state=absent
+
+  #add 'Port 22' entry
+- name: Add port entry - Ubuntu
+  lineinfile: dest=/etc/ssh/sshd_config backup=yes backrefs=yes regexp="^# What ports, IPs and protocols we listen for" line="# What ports, IPs and protocols we listen for\nPort {{ SSH_PORT }}"
+  when: not os_ver.stat.exists
+- name: Add port entry - CentOS
+  lineinfile: dest=/etc/ssh/sshd_config backup=yes backrefs=yes regexp="^# default value." line="# default value.\nPort {{ SSH_PORT }}"
+  when: os_ver.stat.exists
+
```

Comment:
The whole removing the port, then readding the port is un-ansible like. It's also fragile. For example the above fails to remove a line with '  Port' or a line with '# Port'. Matching on a phrase like 'regexp="# What ports, IPs and protocols we listen for"' is also fragile.

For example you can do a regex like so '\bPort\b' will match  '   Port' or '#Port' but not 'LocalPort' etc.